src/main/java/marquez/api/models/JobRequest.java

```diff
@@ -28,12 +30,20 @@ public JobRequest(
       final String type,
       final List<String> inputs,
       final List<String> outputs,
+      final List<DatasetId> inputIds,
+      final List<DatasetId> outputIds,
       @Nullable final String location,
       @Nullable final Map<String, String> context,
       @Nullable final String description) {
     this.type = type;
-    this.inputs = ImmutableList.copyOf(inputs);
-    this.outputs = ImmutableList.copyOf(outputs);
+    if ((inputs != null || outputs != null) && (inputIds != null || outputIds != null)) {
+      throw new IllegalArgumentException(
+          "Either use inputs/outputs or inputIds/outputIds but not both at the same time");
+    }
+    this.inputs = inputs == null ? null : ImmutableList.copyOf(inputs);
+    this.outputs = outputs == null ? null : ImmutableList.copyOf(outputs);
+    this.inputIds = inputIds == null ? null : ImmutableList.copyOf(inputIds);
+    this.outputIds = outputIds == null ? null : ImmutableList.copyOf(outputIds);
     this.location = location;
     this.context = (context == null) ? ImmutableMap.of() : ImmutableMap.copyOf(context);
     this.description = description;
```

Comment:
_Minor_: Like the `DatasetResponse`, let's also return the namespace in the response?